modules/contact/src/userobjects/LMWeightedVelocitiesUserObject.C

```diff
@@ -38,8 +40,12 @@ LMWeightedVelocitiesUserObject::LMWeightedVelocitiesUserObject(const InputParame
                                     ? getVar("lm_variable_tangential_two", 0)
                                     : nullptr)
 {
-  if (!_lm_normal_var || !_lm_variable_tangential_one)
-    paramError("lm_variable_normal", "The Lagrange multiplier variables must be actual variables.");
+  if (isCoupledConstant("lm_variable_tangential_one") || isCoupledConstant("lm_variable_normal"))
+    paramError("lm_variable_normal",
+               "The Lagrange multiplier variable must be an actual variable and not a constant.");
+  else if (!_lm_normal_var || !_lm_variable_tangential_one)
+    paramError("lm_variable_normal",
+               "The Lagrange multiplier variables must be provided and be actual variables.");
```

Comment:
I guess I don't understand the need for this added verbosity. My understanding is that `getVar` should return a `nullptr` if the user has passed a constant in the input file, and so the previous error message should cover that case?

Reply:
I intended to address a comment from Daniel to provide more error granularity, see https://github.com/idaholab/moose/pull/23591#discussion_r1167175127. I can revert it.